addons/mail/static/src/models/activity/activity.js

```diff
@@ -148,8 +148,13 @@ function factory(dependencies) {
                 method: 'activity_format',
                 args: [this.id],
             }));
-            this.update(this.constructor.convertData(data));
-            this.thread.refresh();
+            if (data) {
+                this.update(this.constructor.convertData(data));
+                this.thread.refreshActivities();
```

Comment:
What is the reasoning behind the change from `refresh` to `refreshActivities`?

Reply:
@seb-odoo here, in this case, only activity updation is happening so as now we have a specific function for that. Hence used `refreshActivities` instead of `refresh`.

Reply:
That's not correct. If you mark an activity as done, a message would be added, saying you have completed the task. You need to refresh messages too

![image](https://user-images.githubusercontent.com/48204996/95827967-ae613a00-0d34-11eb-8386-c46fb60444a7.png)

Reply:
See input from xdu. Please revert this change.

Reply:
just do "if -> convert data" "else -> delete" and keep refresh in both cases.

Current PR is about fixing issue when data is not found, it's not about doing anything with refresh.